ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.view.js

```diff
@@ -125,42 +106,115 @@ module.exports = Marionette.LayoutView.extend({
   updateMaxAndMin: function() {
     this.model.setLatLon()
   },
+  isFilterUndefinedOrNull: function(filter) {
+    if (
+      filter === null ||
+      filter.value === undefined ||
+      filter.value === null
+    ) {
+      return true
+    }
+  },
   deserialize: function() {
-    if (!this.propertyModel) return
-
-    const filter = this.propertyModel.get('value')
-    this.model.set(filterToLocationOldModel(filter))
-
-    switch (filter.type) {
-      // these cases are for when the model matches the filter model
-      case 'DWITHIN':
-        if (CQLUtils.isPointRadiusFilter(filter)) {
+    if (this.propertyModel) {
+      var filter = this.propertyModel.get('value')
+      if (this.isFilterUndefinedOrNull(filter)) {
+        return
+      }
+      var filterValue =
+        typeof filter.value === 'object' ? filter.value.value : filter.value
+      switch (filter.type) {
+        // these cases are for when the model matches the filter model
+        case 'DWITHIN':
+          if (CQLUtils.isPointRadiusFilter(filter)) {
+            let pointText = filterValue.substring(6)
+            pointText = pointText.substring(0, pointText.length - 1)
+            var latLon = pointText.split(' ')
```

Comment:
✏️ Shouldn't be mixing `let` and `var`. Should be sticking to `let` when it's volatile or `const` when it should be immutable.